benches/common/mod.rs

```diff
@@ -0,0 +1,22 @@
+use anyhow::{bail, Result};
+use lurk::config::lurk_config;
+use std::collections::HashMap;
+
+pub const PUBLIC_PARAMS_BENCH_PATH: &str = "/var/tmp/lurk_benches/public_params";
+
+/// Sets the config settings with the temporary public parameter cache location
+pub fn set_bench_config() -> Result<()> {
+    let mut overrides = HashMap::new();
+    overrides.insert("public_params_dir", PUBLIC_PARAMS_BENCH_PATH.into());
+    let config = lurk_config(Some(&overrides));
+    // If global config already set, check it has the same
+    // temporary public param cache or return an error
+    if config.public_params_dir == PUBLIC_PARAMS_BENCH_PATH {
+        Ok(())
+    } else {
+        bail!(
+            "Incorrect public parameter cache directory: {}",
+            config.public_params_dir
+        )
+    }
```

Comment:
@huitseeker Is this a good pattern to ensure the config is properly set? It goes against the infallible spirit of `OnceCell::get_or_init`, but it also seems bad to silently write to a permanent place on disk if the `public_params_dir` variable has already been set.

If the latter, should I also return a Result from `lurk_config()` and `cli_config()` so the user can decide whether to continue? It seems less promising for `lurk_config` as it shouldn't have overrides and hence error potential except when used in a binary (as with the above bench config).

Reply:
The issue is that figuring out the correct bench path is hard. Here we force the configuration value to be equal to a hardcoded default, which means I'm confused as to why the config setting exists.

If the trouble is that we don't know how to engineer env-var based overrides from a binary, one answer is [dotenv](https://crates.io/crates/dotenv)

Reply:
Makes sense, I think the better approach is to surface the config in logs so the user can check if the params are stored in the right place, rather than erroring. Anyway, I removed this in favor of the default param path. Thanks!